### ISSUE TITLE
Error: Multiple anonymous defines.

### ISSUE THREAD

**Issue by thanhnguyenminh** [mgechev/angular-seed#1551]
I'm using D3 to setup with Angular seed. Everything is ok on dev evironment but when I build app by using npm run build.prod, I got the message as below

Error on instantiate for d3-color/build/d3-color.js at file:///Volumes/xxxx/projects/xx-xxx/node_modules/d3-color/build/d3-color.js
    Loading d3-ng2-service/src/bundle-d3.js
    Loading d3-ng2-service/src/d3.service.js
    Loading d3-ng2-service/index.js
    Loading dist/tmp/app/app.module.js
    Loading dist/tmp/app/main.js
    Error: Multiple anonymous defines.
    at ParseTreeTransformer.AMDDependenciesTransformer.transformCallExpression (/Volumes/xxxx/projects/xx-xxx/node_modules/systemjs-builder/compilers/amd.js:98:13)

Could you please help to ignore it?

Thanks,
Thanh

**Comment by mgechev** [comment]
Would you share your `SYSTEM_DEV_CONFIG` and `SYSTEM_BUILDER_CONFIG`?

**Comment by aydun1** [comment]
I'm getting the same issue so hopefully I can add some info.

For me at least, `SYSTEM_CONFIG_DEV` and `SYSTEM_BUILDER_CONFIG` are unchanged with the exception of adding the following lines to project.config.ts

`this.SYSTEM_CONFIG_DEV.paths['d3*'] = 
  'node_modules/d3*/index.js';`
` this.SYSTEM_CONFIG_DEV.paths['traceur'] =
      'node_modules/traceur/bin/traceur.js';`

Thanks!

**Comment by aydun1** [comment]
I think all of them..
Besides the above changes, the only changes I've made to a fresh version of this repository are:

In `package.json`:
Added `"@types/d3": "^4.3.0"` to `devDependencies`

Added `"d3": "^4.3.0",` to `dependencies`



And in `src/client/app/app.component.ts`
`import * as d3 from 'd3';`

`constructor() {  
console.log(d3); }`


Network tab filtered for d3 is 449/508 requests. Here's the first page.

![d3](https://cloud.githubusercontent.com/assets/5484467/20126340/6263b02c-a688-11e6-8729-b061a415cb90.PNG)

Thanks so much for your help. Let me know if I can add anything else! Although maybe the issue is more to do with d3?

**Comment by mgechev** [comment]
You need this line in your `project.config.ts`:

```ts
    this.addPackageBundles({
      name: 'd3',
      path: 'node_modules/d3/build/d3.min.js'
    });
```

**Comment by aydun1** [comment]
Awesome, adding that line fixed all the downloads when running with `npm start`!

However, `npm run build.prop` still returns the error about multiple anonymous defines in d3-color.

I tried Both with and without the `this.SYSTEM_CONFIG_DEV.paths['d3*'] = 'node_modules/d3*/index.js';` line (just in case).

I commented out all the define() blocks in `/node_modules/d3-color/build/d3-color.js` and that fixed that error.

Unless I'm doing something wrong, looks to be a problem with that `d3-color` unless I'm doing something wrong. I might start a ticket on d3 unless you have any more insight. Thanks again.
